### PR TITLE
PR: Several things

### PR DESCRIPTION
**Pull request by rlaverde**
I improved the readme, create some cookiecutter variables, and added some templates

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}plugin.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{% endif %}
+
+{% if cookiecutter.create_config_page == 'y' %}
+class ReportsConfigPage(PluginConfigPage):
+    """Reports plugin preferences."""
+    pass
+{% endif %}
+
+class ReportsPlugin(SpyderPluginWidget):
+    """Reports plugin"""
```
Comment (goanpeca) [inline {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}plugin.py]: dot at the end!

**Review by goanpeca — CHANGES_REQUESTED** [review]

File: cookiecutter.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "author": "Spyder Project Contributors",
+  "email": "admin@spyder-ide.org",
+  "github_username": "spyder-ide",
+  "plugin_name": "Demo Plugin",
+  "repo_name": "spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-')}}",
+  "project_name": "{{ cookiecutter.repo_name.replace('-', '_') }}",
+  "description": "Plugin for Spyder IDE.",
+  "version": "0.1.0",
+  "create_config_page": "n",
+  "open_source_license": ["MIT license", "BSD license", "ISC license", "Apache Software License 2.0", "GNU General Public License v3", "Not open source"]
```
Comment (goanpeca) [inline cookiecutter.json]: I don't think we need this. Basically they should be MIT.

File: {{ cookiecutter.repo_name }}/setup.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © {{ cookiecutter.author }}
```
Comment (goanpeca) [inline {{ cookiecutter.repo_name }}/setup.py]: use `(c)`

File: {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/__init__.py [hunk 4]
```diff
@@ -1,21 +1,8 @@
...
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © {{ cookiecutter.author }}
```
Comment (goanpeca) [inline {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/__init__.py]: Use `(c)`

File: {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/tests/__init__.py [hunk 5]
```diff
@@ -1,21 +1,8 @@
...
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © {{ cookiecutter.author }}
```
Comment (goanpeca) [inline {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/tests/__init__.py]: use `(c)`

File: {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/widgets/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}gui.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © {{ cookiecutter.author }}
```
Comment (goanpeca) [inline {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/widgets/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}gui.py]: Use `(c)`

File: {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}plugin.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © {{ cookiecutter.author }}
```
Comment (goanpeca) [inline {{ cookiecutter.repo_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}plugin.py]: Use `(c)`

**Review by goanpeca — CHANGES_REQUESTED** [review]

File: cookiecutter.json (hunk 2 above)
Comment (rlaverde) [reply cookiecutter.json]: But It could be useful if somebody out of spyder developers use it.

And the default value is MIT

Comment (goanpeca) [reply cookiecutter.json]: I mean that they should probably be MIT anyway to keep everything consistent.

File: {{ cookiecutter.repo_name }}/setup.py (hunk 3 above)
Comment (goanpeca) [reply {{ cookiecutter.repo_name }}/setup.py]: @rlaverde ?

File: cookiecutter.json (hunk 2 above)
Comment (rlaverde) [reply cookiecutter.json]: I don't like it, but I'll delete this logic

Comment (goanpeca) [reply cookiecutter.json]: What I am saying is why should we encourage use of any other license if MIT is perfectly fine. Is not like the package they made will be useful for something else besides being a spyder plugin

**Pull request merged**